cli/cmd/install.go

```diff
@@ -717,6 +744,31 @@ func exitIfClusterExists() {
 	os.Exit(1)
 }
 
+// exitIfNamespaceDoesNotExist checks the kubernetes API to determine if the
+// control-plane namespace exists, and returns an error if it does not.
+//
+// This is useful when running `linkerd install control-plane`, where the
+// namespace must exist, but `linkerd-config` should not.
+func exitIfNamespaceDoesNotExist() {
+	hc := newHealthChecker(
+		[]healthcheck.CategoryID{healthcheck.KubernetesAPIChecks},
+		time.Time{},
+	)
+
+	success := hc.RunChecks(exitOnError)
+	if !success {
+		fmt.Fprintln(os.Stderr, "Failed to connect to Kubernetes. If this expected, use the --ignore-cluster flag.")
+		os.Exit(1)
+	}
+
+	err := hc.CheckNamespace(context.Background(), controlPlaneNamespace, true)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Failed to find the Linkerd control-plane namespace. If this expected, use the --ignore-cluster flag.")
```

Comment:
Probably our first suggestion should be to run `linkerd install config`.  Maybe something like: "Failed to find control-plane namespace: %s.  Run `linkerd install config -l %s | kubectl apply -f -` to create it (this requires cluster administration permissions.)  See <website docs link> for more information.  Or use <flag> to proceed anyway."